python/sglang/srt/managers/scheduler.py

```diff
@@ -589,12 +589,15 @@ def handle_generate_request(
         if (
             req.sampling_params.json_schema is not None
             or req.sampling_params.regex is not None
+            or req.sampling_params.ebnf is not None
         ):
             assert self.grammar_backend is not None
             if req.sampling_params.json_schema is not None:
                 key = ("json", req.sampling_params.json_schema)
             elif req.sampling_params.regex is not None:
                 key = ("regex", req.sampling_params.regex)
+            elif req.sampling_params.ebnf is not None:
+                key = ("ebnf", req.sampling_params.ebnf)
 
```

Comment:
Shall we raise a warning when multiple format is specified at the same time?

Reply:
yes we raise an error here for that:
https://github.com/sgl-project/sglang/blob/fda6936b28f8805b02db6064b785b7480697ae32/python/sglang/srt/sampling/sampling_params.py#L116-L122

works like this:
![image](https://github.com/user-attachments/assets/720fc580-f48b-4851-8213-c7c71ddadd07)